lib/matplotlib/figure.py

```diff
@@ -1954,7 +1966,7 @@ def subplot_mosaic(self, mosaic, *, sharex=False, sharey=False,
         per_subplot_kw = self._norm_per_subplot_kw(per_subplot_kw)
 
         # Only accept strict bools to allow a possible future API expansion.
-        _api.check_isinstance(bool, sharex=sharex, sharey=sharey)
+        _api.check_isinstance((bool, str), sharex=sharex, sharey=sharey)
```

Comment:
This handling was intended to to disallow "truthy" (but not `True`, the bool) and "falsy" (but not `False`, the bool) things from giving the True/False case for sharex/y

Since it is happening _after_ normalization to string (on lines 1943-7) this is not actually doing anything here.

I think that normalization should be moved after this check.